tests/core/test_configuration.py

```diff
@@ -557,6 +557,23 @@ def test_command_from_env(self):
             # the environment variable's echo command
             self.assertEqual(test_cmdenv_conf.get('testcmdenv', 'notacommand'), 'OK')
 
+    def test_sensitive_config_values(self):
```

Comment:
The tests should check the behavior, not just the content of the attribute. I cannot imagine a situation when this test could help.

Reply:
Try removing the config from configuration.py and nothing will fail without this test :)  Without it no-one would know why _CMD is not working for sql_alchemy_conn . Hope I changed your imagination ;)

This test tests that these fields are allowed to be run via _CMD or _SECRET.

This PR does not add any functionality and uses existing functionality which is already covered by tests, did you check that?

Reply:
The list changes the behaviour of how config is read and used. So verifying it is important in my opinion

Reply:
I don't have a strong opinion on this, but I would prefer the tests to check the behavior, not just the value in the class attribute. I would be fine if we weren't to assume that we have other tests that cover behaviors, and this is just a configuration option.

In real life, no user will write the code:
````python
config_parser = AirflowConfigParser()
print(config_parser.sensitive_config_values)
````
so this attribute is an internal implementation detail.

Reply:
If we start adding test for each configuration, we will lots of duplication that checks exactly the same behaviour.

Reply:
For example with jinja_templated strings, it is enough that we can render Jinja templates with different kind of inputs. But we should not check the same things for all operators, just checking that the template_field contains the list we expect it to should be enough. Otherwise we will have tons of tests for each_template_field * Number of Operators

Reply:
I agree with @kaxil, the functionality of `_CMD` and `_SECRET` is already covered in tests above.

Reply:
We don't only write tests for what users directly use. We need to test what the expected behaviour of the attribute would be.

> I don't have a strong opinion on this, but I would prefer the tests to check the behavior, not just the value in the class attribute. I would be fine if we weren't to assume that we have other tests that cover behaviors, and this is just a configuration option.
> 
> In real life, no user will write the code:
> 
> ```python
> config_parser = AirflowConfigParser()
> print(config_parser.sensitive_config_values)
> ```
> 
> so this attribute is an internal implementation detail.

Reply:
I'm with @mik-laj  on this -- the test is _tightly_ coupled to the exact implementation detail, meaning even the slightest change (other than formatting) to the code means the test needs  to be updated.

You've essentially just re-written the same code twice here.

Reply:
> template_field contains the list we expect it to should be enough

It is not common.

> Otherwise we will have tons of tests for each_template_field * Number of Operators

In my opinion, it is enough to verify that this behavior only works for one key. We don't need to check that this will work for all keys if the key list is a static list. If something works for one value, it will also work for all other values.

Reply:
> In my opinion, it is enough to verify that this behavior only works for one key. We don't need to check that this will work for all keys if the key list is a static list. If something works for one value, it will also work for all other values.

What he said.

Reply:
Sorry I still disagree. Basically, what you guys are saying we don't add test for this, right?

Reply:
1.10.4 broke for exactly that reason where dataproc operators broke because we didn't have test to verify template_field

>It is not common.

Reply:
I think it is good compromise to not test all. But test the behaviour + the keys we expect it to work with. 

>You've essentially just re-written the same code twice here.

Reply:
Here is the issue: https://issues.apache.org/jira/browse/AIRFLOW-5136 that broke 1.10.4 . We started verifying the list of template fields after that.

Reply:
> Here is the issue: https://issues.apache.org/jira/browse/AIRFLOW-5136 that broke 1.10.4 . We started verifying the list of template fields after that.

Comparing the contents of the `template_fields` field would not help us in detecting this problem. You need to check that `template_fields` field contains the fields that are set in the constructor.

Reply:
This is what I meant:

```
assert DataProcSparkOperator.template_fields == ['arguments', 'job_name', 'cluster_name', 'region', 'dataproc_jars', 'dataproc_properties']
```

Reply:
How is this supposed to prevent aggression? I think during the review, this value will be added to the two arrays and no one will see the difference until they execute the operator.

Reply:
Check the Issue I linked which broke 1.10.4

Also remember, there is a cherry-picking process too which is prone to errors

Reply:
In my opinion, we can merge this PR. This test does not bother me.

Reply:
> This is what I meant:
> 
> ```python
> assert DataProcSparkOperator.template_fields == ['arguments', 'job_name', 'cluster_name', 'region', 'dataproc_jars', 'dataproc_properties']
> ```

On this one I must agree with Kamil. This test is useless. This test will work even if someone renames `self.job_name` to `self.magic_dataproc_job_name`. The proper test would be like:
```py
assert all(t in op.__dict__ for t in op.template_fileds)
```
the problem if I'm not mistaken arises when there's difference between `template_fields` and operator attributes.

Reply:
:D Maybe a badly chosen example in that case. Let's agree to dis-agree on this one.